static/js/tab_bar.js

```diff
@@ -131,10 +131,10 @@ exports.exit_search = function () {
     }
 };
 
-exports.update_stream_name = function (new_name) {
+exports.update_stream_name = function () {
     const stream_name = $(".stream a");
     if (stream_name !== undefined) {
-        stream_name.text(new_name);
+        this.initialize();
```

Comment:
This probably is wrong -- if you read `initialize`, it sets up a click handler that we don't want.  It seems like we should call `build_tab_bar` instead?

Reply:
I guess it might happen to work, but we should refactor `tab_bar.initialize` to have a cleaner interface.

Reply:
Could you explain a little?